### ISSUE TITLE
QGIS icon missing in GNOME 3 applications menu

### ISSUE THREAD

**Issue by das-g** [NixOS/nixpkgs#31199]
## Issue description

After installing QGIS, the QGIS icon isn't in the GNOME 3 application menu.

### Steps to reproduce

1. Install QGIS and GNOME 3
   I did so by this `/etc/nixos/configuration.nix`:
   <details><summary>(click for full file content)
   
   ```nix
   { config, pkgs, ... }:
   {
   # ...
     services.xserver.desktopManager.gnome3.enable = true;
   # ...
     environment.systemPackages = with pkgs; [
       # ...
       qgis
       # ...
     ];
   # ...
   ```
   </summary>

   ```nix
   # Edit this configuration file to define what should be installed on
   # your system.  Help is available in the configuration.nix(5) man page
   # and in the NixOS manual (accessible by running ‘nixos-help’).
   
   { config, pkgs, ... }:
   
   {
     imports =
       [
         # Include the results of the hardware scan.
         ./hardware-configuration.nix
   
         # Workaround from
         # https://github.com/NixOS/nixpkgs/issues/14318#issuecomment-309250231
         ./localectl-fix.nix
       ];
   
     boot.loader.grub.device = "/dev/sda";
     boot.initrd.luks.devices = [
       {
         name = "lvm";
         device = "/dev/sda2";
       }
     ];
   
     # Use the systemd-boot EFI boot loader.
     boot.loader.systemd-boot.enable = true;
     boot.loader.efi.canTouchEfiVariables = true;
   
     # Select internationalisation properties.
     i18n = {
       consoleFont = "Lat2-Terminus16";
       consoleKeyMap = "de_CH-latin1";
       defaultLocale = "de_CH.UTF-8";
     };
   
     # Set your time zone.
     time.timeZone = "Europe/Zurich";
   
     # List packages installed in system profile. To search by name, run:
     # $ nix-env -qaP | grep wget
     environment.systemPackages = with pkgs; [
       git
       git-cola
       nix-repl
       gnupg
       firefox
       thunderbird
       qgis
       libreoffice
       tdesktop
     ];
   
     # Enable the X11 windowing system.
     services.xserver.enable = true;
     services.xserver.layout = "ch,de";
     services.xserver.displayManager.gdm.enable = true;
     services.xserver.desktopManager.gnome3.enable = true;
     services.xserver.localectlFix.enable = true;
   
     users.extraUsers.das-g = {
       isNormalUser = true;
       home = "/home/das-g";
       description = "Raphael Das Gupta";
       extraGroups = ["wheel" "networkmanager"];
     };
     programs.vim.defaultEditor = true;
     programs.command-not-found.enable = true;
   
     # This value determines the NixOS release with which your system is to be
     # compatible, in order to avoid breaking some software such as database
     # servers. You should change this only after NixOS release notes say you
     # should.
     system.stateVersion = "17.09"; # Did you read the comment?
   
   }
   ```
   </details>

2. rebuild and switch
   (To work around #21439 I did so by
   ```bash
   sudo nixos-rebuild boot --upgrade
   sudo reboot
   ```
   )
3. log into GNOME 3 as normal user
4. open the applications menu (move mouse pointer to upper left corner or press meta key)

### Expected
* When now typing "qgis" or "QGIS" or similar into the search field, the QGIS icon should appear.
* When (instead of typing something) clicking the "Show Applications" icon (the 3&times;3 dots) and then the "All applications" button, the QGIS icon should be among the listed ones

### Observed
* QGIS doesn't appear in the GNOME application menu, even when scrolling through all pages of the "All applications" view
* It can still be started by running the `qgis` command (in a terminal window or with <kbd>Alt</kbd>+<kbd>F2</kbd>)
  * when you do that, sometimes you get the error message window
    > Aktualisierung der Sicht in privater qgis.db gescheitert.
    > table vw_srs already exists

   and, after clicking "OK" on that, in the terminal:
   > Warning: QCss::Parser - Failed to load file  "/style.qss" 
   > QInotifyFileSystemWatcherEngine::addPaths: inotify_add_watch failed: No such file or directory
   > Warning: QFileSystemWatcher: failed to add paths: /home/das-g/.qgis2//project_templates
   > Warning: QLayout: Attempting to add QLayout "" to QgsPanelWidgetStack "mWidgetStack", which already has a layout

    (Whether that is related, I don't know. I'll file it as a separate issue)

## Technical details

* System: NixOS: `17.09.1927.6a0a00d91e (Hummingbird)`
* Nix version: `nix-env (Nix) 1.11.15`
* Nixpkgs version: `"17.09.1927.6a0a00d91e"`
* Sandboxing enabled: `build-use-sandbox = false`

**Comment by das-g** [comment]
@viric, you're listed as the package's maintainer. Can you help here?

**Comment by jtojnar** [comment]
The issue is that QGIS does not install a desktop file. We would need to copy what Arch is doing:

https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/qgis&id=118f28ad41a686a4067848438aeeff08edda397e#n90

**Comment by lsix** [comment]
@das-g as far as I can see, this is fixed with the update to QGIS 3. Can you confirm that with the new version, you do not have the issue anymore?

**Comment by das-g** [comment]
Now that I could test this on NixOS 19.03, I can confirm that the issue is indeed solved: There is now a launcher e.g. in the gnome 3 application search.
